paperqa/agents/main.py

```diff
@@ -51,7 +51,7 @@ async def agent_query(
     if docs is None:
         docs = Docs()
 
-    search_index = SearchIndex(
+    answers_index = SearchIndex(
```

Comment:
What's the distinction between a search and answer index? You have `search_or_answers_index` below too

Reply:
Yeah I actually made this change because I had a similar question in my head. It turns out we instantiate two types of `SearchIndex`:

- "answer" index: we hardcode `index_name="answers"` and store only `Answer.answer`s inside
- "search" index: we don't hardcode the `index_name`, and use this for normal documents

Before this PR, everything was named `search_index` to match the type `SearchIndex`, but this name lost the semantics of the index (e.g. one is for searching, one is for housing answers)

Reply:
So, do you think I should do something different here? I have `_index` to match the type `SearchIndex`, and the prefix `answer` or `search` to align with its semantics

Reply:
Ah ok, I missed `index_name="answers"`. Thanks for explaining. I think this makes sense, though a comment by `search_or_answers_index` might help.

Reply:
I agree that `search_or_answers_index` leaves one wonder a bit. In the case of `search_or_answers_index`, it's being made to `query` and then immediately discard. So I renamed it to `index_to_query`